ecosystem/sep-0038.md

```diff
@@ -0,0 +1,200 @@
+
+Status Code | Name | Reason
+-----|------|------------
+`400` | Bad Request | The request is invalid in any way.
+`403` | Permission Denied | No `Authorization` header has been provided or the contents of the header are not accepted as valid.
+
+The response body must contain a human-readable description of the reason for error:
+
+```json
+{
+  "error": "The requested asset is not supported. See GET /prices for supported assets."
+}
+```
+
+### Asset Identification Format
+
+This protocol can be used to provide quotes for any class of asset in exchange for a Stellar asset. The following format is used to identify an asset in API requests and responses.
+
+```
+<scheme>:<identifer>
+```
+
+The currently accepted `scheme` values are:
+
+Name | Description
+-----|------------
+`stellar` | Used for Stellar assets.
+`iso4217` | Used for fiat currencies.
+
+`stellar` Identifier Format
+
+The [SEP-11](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset) asset format.
+
+`iso4217` Identifier Format
+
+The [ISO 4217](https://www.iso.org/iso-4217-currency-codes.html) code for the fiat currency.
+
+### Endpoints
+
+* [`GET /info`](#get-info)
+* [`GET /prices`](#get-prices)
+* [`GET /price`](#get-price)
+* [`POST /quote`](#post-quote)
+
+### GET Info
+
+This endpoint describes the supported Stellar assets and off-chain assets available for trading. Note that the anchor may not support a trading pair between every Stellar asset and off-chain asset listed. Use the [`GET /prices`](#get-prices) endpoint to see which pairs are supported.
+
+#### Request
+
+No request arguments required.
+
+#### Response
+
+Name | Type | Description
+-----|------|------------
+`assets` | array | An array of objects describing the assets available in exchange for one or more of the other assets listed.
+
+`assets` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`asset` | string | The [Asset Identification Format](#asset-identification-format) value.
+`delivery_methods` | array | (optional) Only for non-Stellar assets. An array of objects describing the methods a client can use to deliver funds to the anchor. The method of delivery may affect the expiration and/or price provided in a [`POST /quote`](#post-quote) response.
+
+`delivery_methods` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`name` | string | The value to use when making `POST /quote` requests.
+`description` | string | A human readable description of the method identified by `name`.
+
+### GET Prices
+
+This endpoint can be used to fetch the [indicative](https://www.investopedia.com/terms/i/indicativequote.asp) prices of available off-chain assets in exchange for a Stellar asset and vice versa.
+
+#### Request
+
+Name | Type | Description
+-----|------|------------
+`sell_asset` | string | The asset you want to sell, using the [Asset Identification Format](#asset-identification-format).
```

Comment:
❓ Do you think we need to support `buy_asset` too? For example, what if I want to buy off-chain asset A, and I don't care what asset I have because I could use path payments to give you whatever asset you're willing to receive.

Reply:
That is a possibility, but in this case the client would be at the whim of the rates offered on the DEX between the asset held and the asset the anchor expects to receive. For context, most (if not all) SEP-31 sending anchors do not use path payments for this reason, and opt to instead hold a float of the destination asset instead.

This case is also only realistic for the on-chain-to-off-chain flow, since clients likely only have one fiat currency (or any asset) off-chain that they're interested in exchanging for `buy_asset`.

Reply:
Got it. I was just curious. This is something that could be added if we needed it, but we'd be better off leaving it out if there is no convincing evidence it'd be used. Thanks.